msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -0,0 +1,462 @@
+---
+title: Create API based Message extension
+author: v-ypalikila
+description: Learn how to build an API message extension and API plugin using Teams developer portal and Teams Toolkit.
+ms.localizationpriority: medium
+ms.topic: overview
+ms.author: anclear
+ms.date: 09/07/2023
+---
+
+# API based Message extension
+
+> [!NOTE]
+> API based message extension only supports search commands.
+
+API based message extensions use a web service to handle user requests and responses. They don't require a bot registration or a bot framework SDK. They can be configured and deployed using the Developer Portal for Teams or the Teams Toolkit.
+
+You can create an API based message extension in Teams using an OpenAPI Specification document. After you've created an OpenAPI Specification document for the APIs you want to use, upload the OpenAPI Specification document to Teams Toolkit or Developer portal for Teams to generate and integrate the client code in your app's project. Create or generate an Adaptive Card template to handle the responses from the API.
+
+An API based message extension helps to interact directly with third-party data, apps, and services, enhancing its capabilities and broadening its range of capabilities. With APIs for message extension, you can:
+
+* Retrieve real-time information, for example, latest news coverage on a product launch.
+* Retrieve knowledge-based information, for example, my team’s design files in Figma.
+* Perform actions on behalf of the user, for example, create a Jira ticket.
+
+## Prerequisites
+
+1. **OpenAPI Specification (OAS)**: Defines a standard, language-agnostic interface to HTTP APIs, which allows both humans and computers to discover and understand the capabilities of the service without access to source code, documentation, or through network traffic inspection. An OpenAPI Specification is used by documentation generation tools to display the API, code generation tools to generate servers and clients in various programming languages, testing tools, and many other use cases.
```

Comment:
Maybe clarify that the prereq here is to have an existing API that is defined by OAS.

Reply:
Updated.